### ISSUE TITLE
Volume companion module

### ISSUE THREAD

**Issue by Zac9080** [Mahmud0808/Iconify#666]
### Describe the bug

Firstly - Iconify version 6.9.0 magisk module for volume causing phone stuck in " Phone is starting " screen causing bootloop. While version iconify 6.8.0 works perfectly. Can someone guide me to fix this issue.

Secondly - How to change icons color system wide. It only allow changes icon color in settings only.

### Expected behavior

Module causing phone to freeze

### Device information

- Device: [e.g. Samsung galaxy S8 plus]
- OS: [lneage os 21 - Android 14]
- App version: [e.g. v6..9.0]
- Magisk/KernelSU: [e.g. Magisk manager 28001]

**Comment by Mahmud0808** [comment]
I won't suggest anyone to use volume module. It hasn't been maintained for a long time, last updated was for maybe android 13. But I didn't hide it as it still works for some people on android 14. Probably one of your rom update made it incompatible as I don't remember changing anything in volume module between v6.8 and v6.9

What icon color are you talking about if not settings icon?

**Comment by Zac9080** [comment]
Understood. Yeah I meant like font color can be changed by creating a custom font magisk module through zfont tool app. Like wise for icons to change color. Something like zfont

**Comment by Mahmud0808** [comment]
No, that can not be done. Icon color is controlled by SystemUI and forcing it to show a different color will have bad side effects.

**Comment by SMOLKEYS** [comment]
> I won't suggest anyone to use volume module. It hasn't been maintained for a long time, last updated was for maybe android 13.

I would personally recommend showing a warning when using the volume slider module on A14. I fell victim to the complete failure of SystemUI because of it and lost data I didn't have backed up.

Alternatively, you could also make an entire special category for the more riskier customization options and put the volume module there.

**Comment by AsLoLoks** [comment]
Same problem. By i restored my phone until i installed iconify again and imported config file. (accept for module, IDK this) If volume are store there don't import it, if not you probably better with saving what you can and wiping your phone. Idk 
the implications of this bug and it is not a fix for it.

Instructions for making phone "usable" (with few unnecessary steps probably)
Boot in safe mode.
Find and uninstall apps with iconify in name
Reboot Normall
Remove companion module and iconify form magisk
Reboot
Re enable Zygisk and other modules 
Reboot 
Now in theory you phoned works how know for how long.
To boot in save mode in power menu hold power off or restart button. Or google how to do this for your device.

**Comment by SMOLKEYS** [comment]
> If volume are store there don't import it

I mean, if you have a config file that contains volume slider modifications, you *can* import it just fine, as long as you don't have (or disabled) the volume slider companion module, which seems to be the one causing the damage.

**Comment by SMOLKEYS** [comment]
The only safe volume slider modifications from what I've seen are the "Volume Slider Track" and the Volume Panel options in the XPosed category.

**Comment by AsLoLoks** [comment]
> you can import it just fine, as long as you don't have (or disabled) the volume slider companion module

I was thinking the same thing because if I'm already creating a standalone module for it why should it have any connection to config file but importing it just crushed my phone and got me to android corrupted screen and i had companion modules uninstalled and rebooted like 3 times.

**Comment by mxKeaton** [comment]
This just took a whole day of trial and error to revert and not loose data. Please add a disclaimer or a warning!

**Comment by mxKeaton** [comment]
> Instructions for making phone "usable" (with few unnecessary steps probably) Boot in safe mode. Find and uninstall apps with iconify in name Reboot Normall Remove companion module and iconify form magisk Reboot Re enable Zygisk and other modules Reboot Now in theory you phoned works how know for how long. To boot in save mode in power menu hold power off or restart button. Or google how to do this for your device.

@AsLoLoks 
I messed up again and the issue reintroduced itself, whilst my previous fix does not work (deleting folders via TWRP & sideloading fresh OS). Now I wanted to try your approach, but booting into safe mode does not bring me far. I have the issue that the phone boots regularly now and it lets me log into my sim, but then it flickers and crashes, same in safe mode. The module folder is already empty and the issue still persists. Any ideas?

**Comment by Mahmud0808** [comment]
Added warning on it for android 14 and above.
